### PR TITLE
Add RPC for starting and terminating a connection

### LINKED ISSUE #1113: [FEATURE] Add RPCs for creating and terminating a connection to server
Will include implementation of the following methods specified in RFC10, along with the basic machinery to set up a connection over gRPC:

```scala
object TransEx {
  /**  Create a transition explorer
   *
   * Establishes the connection and a running session
   *
   * The channel is managed by the gRPC framework. */
  def apply(channel: Channel): TransEx = ...
}

  /** Terminate the session. */
  def terminate(): Unit
```

This work item will include

- [x] Add [ScalaPB](https://scalapb.github.io/docs/installation) as a dependency
- [x] Defining the basic data structure to track connections
- [x] Creating the protobuf schema for this data structure
- Add testing harness: #1848 
- Expose server start to CLI: #1848
- Adding a test using the RPC system to create a connection #1848

If the work on this ends up exceeding ~400loc, I'll refactor into two separate PRs.

### PR DESCRIPTION
**Pull request by shonfeder**
Towards #1113

## What's added

This changeset makes the following additions: 

- Adds the various dependencies for the gRPC stack
- Adds a new project, `hai`, to group the server functionality
- Adds the start of the protobuf spec, defining a `TransExplorer` service with a single RPC for opening a connection with the server.
- Implements the minimal service and server functionality classes
- Documents the basic stack, organization, and concepts upon which the server development will proceed.

## How to review

The commits are well organized, and reviewing by commit will probably be quite easy. 90% of of the work is in fc53bcfb2b7e56a66f4731a9648f52c8b92510e3.

## Caveats

No tests are added here yet, as meaningful tests will require implementation of the client, at least. But I think there is enough here to get feedback already, and provide a digestible chunk to onboard reviewers.

---

- [ ] ~~Tests added for any new code~~
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality

### CODE REVIEW

**Comment by shonfeder** [comment]
I've tagged in a wider audience than necessary for review, just because this is a good changeset to look over to onboard onto the basic set up. I think if you understand the basic pieces here, it will be easy to filter out the RPC noise in future changes and just see the relevant business logic. I'm not expecting/asking everyone to spend a bunch of time reviewing tho :)

**Review by konnov — APPROVED** [review]
Looks good, except the project name reminds me of Thanh Hai's name all the time. Maybe we should add one more letter in the abbreviation, so it does not fully match with his name. For example: hapi, haint, hmci, etc.

File: hai/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Hai: Human-Apalache Interaction
```
[the remark this answers is not in the document]
Comment (shonfeder) [reply hai/README.md:1]: Could we take that as another good reason **for** this name? We could add to the note on meanings that it is also in honor of one of the original authors?

**Comment by shonfeder** [comment]
Changed the name. Thanks for the review!

**Pull request merged**